### ISSUE TITLE
Windows issue(keyboard ,output and linspace problem) 

### ISSUE THREAD

**Issue by aburousan** [MathInspector/MathInspector#4]
I was trying out the program with same things you have shown in your latest video. But I noticed few problems.

1.  First was "d" is not working. I thought it's y keyboard problem but no, that program is doing something strange as shown below,
![d_issue](https://user-images.githubusercontent.com/60939745/106155237-74625980-61a6-11eb-9036-00f65f7f96ff.png)

2. Second was the issue of disconnecting the output. I have to click on the objects and click on disconnect to remove them from output. Although i can disconnect one function from other using mouse.
3.  The most important one is the error of **linspace(don't know if others have similar problems)** . In the **node editor**, the input nodes are not shown for linspace as shown 
![Screenshot (20)](https://user-images.githubusercontent.com/60939745/106156068-1bdf8c00-61a7-11eb-8210-77bef328aef1.png)
4. Also the plot option is missing as shown
![Screenshot (21)](https://user-images.githubusercontent.com/60939745/106156195-416c9580-61a7-11eb-975e-770c09b671b9.png)
5. Also there are some pygame error, i don't know why but a blank **pygame** window is created after trying out domaincoloring for few times(although it is working fine) and i can't close that.
6.  There is also an issue with opening new file in module. It can't open the program file in submile text3 .
This software is too powerful and i really love it. I hope these things will be resolved quickly as i can't wait to make a video using this.

**Comment by GAustinCorbett** [comment]
I'm also having the issue with the letter "d".

Also, numpy is automatically imported on startup.

**Comment by etowahs** [comment]
I did not have issues with the letter "d." I also found that numpy is automatically imported. 
![image](https://user-images.githubusercontent.com/14843470/106392338-4b68e100-63bf-11eb-905d-0f08ded9e6c4.png)

However, it looks like numpy was not properly imported since the linspace function was still not defined.

**Comment by MathInspector** [comment]
Thanks for this issue report!  Apologies for the delay in responding.  I believe the issue with ctrl+d should be resolved now.

See this commit for more info.  It was trying to replicate the ctrl+d word highlighting hotkey of sublime text, but there is an issue with properly detecting the ctrl modifier key.
https://github.com/MathInspector/MathInspector/commit/da34e2a33d8e09adfce4a46bcc4382a19468a03e

@etowahs yes that was a problem in a previous build, which should have been resolved a couple of days ago.  See here for the fix: https://github.com/MathInspector/MathInspector/commit/0c43ab9ba6deb3a9a0d613522d57282a9f117336
Basically what happened is windows couldn't find the frozen version of numpy, so I redid the logic for avoiding recursive imports.

I will leave this issue open for a few days to confirm these fixes really work.  Please let me know if there is still a problem

**Comment by ryanizer** [comment]
I just downloaded and installed Math Inspector 0.9.0 (Beta).

I'm seeing linspace with no inputs as well. The context menu is not what shows up in the most recent youtube video either.

![image](https://user-images.githubusercontent.com/1178152/106639193-5439ee00-654a-11eb-8605-c851324c82ce.png)

**Comment by ryanizer** [comment]
I recommend incrementing some sort of version number when you make a new build, not doing so makes things harder than they need to be.

**Comment by MathInspector** [comment]
Good call on the version number for the installer file.  The reason I was leaving that out was to find out if windows smart screen will stop preventing people from installing the app based on the filename (the idea being if I keep changing the filename, smart screen will take more time to build up reputation).  Not sure if this is how smart screen works or not, but read somewhere that might help.  From now on I will include version numbers in the installer file.

I'm working on the fixes for everything in this issue, and hopefully before the end of the week they will all be resolved

**Comment by ryanizer** [comment]
Makes sense, maybe just use a global version number that gets incremented with ANY change, installer or core code? That way I can refer to a single number that is shown in the console when I start your amazing app. Great work, btw!

**Comment by MathInspector** [comment]
I  have just uploaded a new version 0.9.1 to the website (https://mathinspector.com/download.html) and pushed the latest code to the master branch.  The download page now has a list of changes in the new version, and the version number is part of the installer file.

1. d key should be working now
2. I believe this might be fixed now.  I suspect what you were doing was hovering and dragging from the output node on the right hand side, instead of the wire coming out of the object.  I made a change in this version so that its possible to disconnect the wire from the output in this way.  Let me know if this is working for you in the latest version or not
3. I am confident the issue with linspace not having input nodes should be fixed in the latest version.  I I was able to reproduce this one, and tested the fix a bunch of times on my machine to confirm.
4. When functions don't have sufficient arguments, then it says "Show Output" instead of "Show Plot".  In this case, since linspace had a bug where the arguments were not showing up, it was saying "Show Output".  Once you put the arguments into the linspace item in the node editor, the right click menu will say "Show Plot"
5. This can happen when pygame crashes out.  I once saw a similar bug where the window would reopen.  I have a feeling I know what went wrong, it had to do with setting the plot config options.  I made some changes that will help hopefully.  Let me know if you are continuing to see this kind of thing.
6. Opening files is done with the system command "start" on windows.  This will try and find the default program you have set for editing files with a given extension.  To resolve this issue, I think you will need to configure your system so that sublime text is the default program for opening files with a .py extension.

Let me know if these fixes worked for you, and if so we can close this issue.  Thanks so much for helping to test and reporting this!

**Comment by Rousan99** [comment]
Thank you!!!.... I will try that out and see if they are alright

**Comment by MathInspector** [comment]
I am going to close this issue for the sake of keeping the issue section well organized, since I believe everything you mentioned has been resolved.  Feel free to open additional issues, or to reopen this one if any of these issues persist.  Cheers =)
